client/src/main/java/io/pravega/client/batch/BatchClient.java

```diff
@@ -0,0 +1,96 @@
+ * {@link EventStreamWriter#writeEvent(String, Object)}. Events within a routing key are strictly
+ * ordered (i.e. They must go the the same reader or its replacement). However because
+ * {@link ReaderGroup}s process events in parallel there is no ordering between different readers.
+ * 
+ * <p>
+ * A note on scaling: Because a stream can grow in its event rate, streams are divided into
+ * Segments. For the most part this is an implementation detail. However its worth understanding
+ * that the way a stream is divided between multiple readers in a group that wish to split the
+ * messages between them is by giving different segments to different readers.
+ */
+@Beta
+public interface BatchClient {
+
+    /**
+     * Returns metadata about the requested stream.
+     * 
+     * @param stream The stream
+     * @return The metadata for the provided stream
+     */
+    StreamInfo getStreamInfo(Stream stream);
+
+    /**
+     * Provides a list of segments and their metadata for a given stream.
+     * 
+     * @param stream the stream
+     * @return The segments in the requested stream.
+     */
+    Iterator<SegmentInfo> listSegments(Stream stream);
+
+    /**
+     * List all the segments from the provided stream that contain any entries after the date listed.
+     * 
+     * @param stream The stream whos segments to list.
+     * @param from The time after which segments should be returned
+     * @return All segments that contain some data on the requested stream after the specified time.
+     */
+    Iterator<SegmentInfo> listSegments(Stream stream, Date from);
+
+    /**
+     * Provides a SegmentIterator to read the events in the requested segment starting from the
+     * beginning of the segment and ending at the current end of the segment.
+     * 
+     * @param <T> The type of events written to the segment.
+     * @param segment The segment to read from
+     * @param deserializer A deserializer to be used to parse events
+     * @return A SegmentIterator over the requested segment
+     */
+    <T> SegmentIterator<T> readSegment(Segment segment, Serializer<T> deserializer);
+
+    /**
+     * Provides a SegmentIterator to read the events after the startingOffset but before the
+     * endingOffset in the requested segment.
+     * 
+     * Offsets can be obtained by calling {@link SegmentIterator#getOffset()} or {@link SegmentInfo#getLength()}
+     * 
+     * @param <T> The type of events written to the segment.
+     * @param segment The segment to read from
+     * @param deserializer A deserializer to be used to parse events
+     * @param startingOffset The offset to start iterating from.
+     * @param endingOffset The offset to stop iterating at.
+     * @return A SegmentIterator over the requested segment at startingOffset
+     */
+    <T> SegmentIterator<T> readSegment(Segment segment, Serializer<T> deserializer, long startingOffset, long endingOffset);
```

Comment:
How do we expect the application to obtain the offsets to use as starting and ending?

Reply:
From SegmentIterator.getOffset or SegmentInfo.length

Reply:
Is this a byte API? I assume not, so would it make more sense to request some sort of "position" object instead of offsets? If someone starts reading at an offset in the middle of an event, how would we handle that?